autogpt_platform/backend/backend/data/model.py

```diff
@@ -206,20 +207,35 @@ def bearer(self) -> str:
 class APIKeyCredentials(_BaseCredentials):
     type: Literal["api_key"] = "api_key"
     api_key: SecretStr
-    expires_at: Optional[int]
+    expires_at: Optional[int] = Field(
+        default=None,
+        description="Unix timestamp (seconds) indicating when the API key expires (if at all)",
+    )
     """Unix timestamp (seconds) indicating when the API key expires (if at all)"""
 
     def bearer(self) -> str:
         return f"Bearer {self.api_key.get_secret_value()}"
 
 
+class UserPasswordCredentials(_BaseCredentials):
+    type: Literal["user_password"] = "user_password"
+    username: SecretStr
+    password: SecretStr
+
+    def bearer(self) -> str:
```

Comment:
since this isn't actually a `Bearer` thing:
```suggestion
    def authorization_header(self) -> str:
```
or
```suggestion
    def auth_basic(self) -> str:
```

Reply:
I tried this but the bearer is actually used various places so its kinda annoying. I fixed it for now though